### DISCUSSION TITLE
Moving components

### DISCUSSION THREAD

**Discussion by yrashk** [ArthurSonzogni/FTXUI#680] [Q&A] [answered]
Suppose I want to make movable/draggable components. I easily implemented an event handler that listens for mouse events, and tracks pressed buttons, and can set the moving flag and get the coordinates. I can also track the relative movement by tracking event's `cursor_x()` and `cursor_y()`.

What I don't yet feel like I understand (or even possible at the moment), is how to tell FTXUI to use these computed coordinates.

Any idea how to do this? If it is impossible, do you think it may be considered?

**Comment by ArthurSonzogni** [comment, marked as answer]
Yes. That's a good idea. Thanks!
Not easy at all, but I might be willing to give it a try. I opened an issue to track this:
https://github.com/ArthurSonzogni/FTXUI/issues/682

**Reply by ArthurSonzogni** [reply]
Implemented: https://github.com/ArthurSonzogni/FTXUI/commit/e19550ae69eedd720f47e1d4911d5119b22838e3
